src/components/TimeBar.tsx

```diff
@@ -45,16 +52,43 @@ const TimeBar: React.FC<TimeBarProps> = ({ targetFPS = 60, startColor = green, e
   };
 
   const finish = (color?: RGBColor) => {
+    const avgTimeUsed =
+      percentTimeLeftWhenRestarted!.reduce((prev, curr) => prev + curr, (currWidth.current / clientWidth) * 100) / (restartCount! + 1);
+    setAverageTimeUsed(avgTimeUsed);
+    setPercentTimeLeftWhenRestarted(undefined);
+    setRestartCount(undefined);
     clearInterval(loop!);
     currWidth.current = clientWidth;
     color && (barRef.current!.style.backgroundColor = `rgb(${color.red}, ${color.green}, ${color.blue})`);
   };
 
+  const avg = useCallback(() => {
+    if (level == Levels.BEGINNER) console.log('avg && beginner');
+    if (restartCount == undefined) {
+      setRestartCount(0);
+    } else {
+      setRestartCount(restartCount + 1);
+    }
```

Comment:
if we do that, then it will add a restart when the player placed the first rectangle, and divide averageTimeUsed with averageTimeUsed.Length + 1, which would give a wrong average